modules/currency-l1/src/main/scala/org/tessellation/currency/l1/Main.scala

```diff
@@ -0,0 +1,177 @@
+import org.tessellation.dag.l1.http.p2p.P2PClient
+import org.tessellation.dag.l1.infrastructure.block.rumor.handler.blockRumorHandler
+import org.tessellation.dag.l1.modules._
+import org.tessellation.dag.l1.{DagL1KryoRegistrationIdRange, StateChannel, dagL1KryoRegistrar}
+import org.tessellation.ext.cats.effect.ResourceIO
+import org.tessellation.ext.kryo.{KryoRegistrationId, MapRegistrationId}
+import org.tessellation.schema.cluster.ClusterId
+import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.node.NodeState.SessionStarted
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.app.{SDK, TessellationIOApp}
+import org.tessellation.sdk.infrastructure.gossip.{GossipDaemon, RumorHandlers}
+import org.tessellation.sdk.resources.MkHttpServer
+import org.tessellation.sdk.resources.MkHttpServer.ServerName
+import org.tessellation.sdk.{SdkOrSharedOrKernelRegistrationIdRange, sdkKryoRegistrar}
+
+import com.monovore.decline.Opts
+import eu.timepit.refined.boolean.Or
+import fs2.Stream
+
+object Main
+    extends TessellationIOApp[Run](
+      s"Currency-l1",
+      s"Currency L1 node",
+      ClusterId(UUID.fromString("517c3a05-9219-471b-a54c-21b7d72f4ae5")),
+      version = BuildInfo.version
+    ) {
+
+  val opts: Opts[Run] = cli.method.opts
+
+  type KryoRegistrationIdRange = CurrencyKryoRegistrationIdRange Or SdkOrSharedOrKernelRegistrationIdRange Or DagL1KryoRegistrationIdRange
+
+  val kryoRegistrar: Map[Class[_], KryoRegistrationId[KryoRegistrationIdRange]] =
+    currencyKryoRegistrar.union(sdkKryoRegistrar).union(dagL1KryoRegistrar)
+
+  def run(method: Run, sdk: SDK[IO]): Resource[IO, Unit] = {
+    import sdk._
+
+    val cfg = method.appConfig
+
+    for {
+      queues <- Queues.make[IO, CurrencyTransaction, CurrencyBlock](sdkQueues).asResource
+      currencyStorages <- CurrencyStorages
+        .make[IO, CurrencyTransaction, CurrencyBlock, CurrencySnapshot](sdkStorages, method.l0Peer, method.globalL0Peer)
+        .asResource
+      validators = Validators.make[IO, CurrencyTransaction, CurrencyBlock, CurrencySnapshot](currencyStorages, seedlist)
+      p2pClient = P2PClient.make[IO, CurrencyTransaction, CurrencyBlock](sdkP2PClient, sdkResources.client, currencyPathPrefix = "currency")
+      services = Services
+        .make[IO, CurrencyTransaction, CurrencyBlock, CurrencySnapshot](
+          currencyStorages,
+          currencyStorages.lastGlobalSnapshot,
+          currencyStorages.globalL0Cluster,
+          validators,
+          sdkServices,
+          p2pClient,
+          cfg
+        )
+      snapshotProcessor = CurrencySnapshotProcessor.make(
+        method.identifier,
+        currencyStorages.address,
+        currencyStorages.block,
+        currencyStorages.lastGlobalSnapshot,
+        currencyStorages.lastSnapshot,
+        currencyStorages.transaction
+      )
+      programs = CurrencyPrograms.make(sdkPrograms, p2pClient, currencyStorages, snapshotProcessor)
```

Comment:
shouldn't it be just `Programs`, like we do with sdk and dag?

Reply:
I guess it can be just `Programs`. Probably the reason I did it that way was that some of the modules from abstraction could be reused as they were, and possible to emphasize that these one's here actually come from currency-l1 module I might have added the prefix. Will remove it.

Reply:
Actually the reason might have been that the complier perceived it as a cyclic dependency, e.i. `Programs` extending `Programs` (from dagL1), but I will handle it with aliasing when imporitng.